src/bin/pg_autoctl/cli_common.c

```diff
@@ -1279,3 +1279,25 @@ stop_postgres_and_remove_pgdata_and_config(ConfigFilePaths *pathnames,
 		exit(EXIT_CODE_BAD_CONFIG);
 	}
 }
+
+
+/*
+ * cli_pg_autoctl_reload signals the pg_autoctl process to reload its
+ * configuration by sending it the SIGHUP signal.
+ */
+bool
+cli_pg_autoctl_reload(const char *pidfile)
+{
+	pid_t pid;
+
+	if (read_pidfile(pidfile, &pid))
+	{
```

Comment:
Maybe do a sanity check that the pid is positive otherwise you can accidently send SIGHUP to all processes (I remember seeing a check like this somewhere else in the codebase):
https://rachelbythebay.com/w/2014/08/19/fork/

Reply:
Done now!